### PR TITLE
feat: website link component defaults target to _blank when the href is external

### PR DESCRIPTION
**Pull request by gobengo**
Motivation:
* CIC - this was at top of CIC backlog for the repo
* https://github.com/web3-storage/web3.storage/issues/2006

How?
* made it so the go-to website `<Link />` component will enforce the logic @jchris requested

### CODE REVIEW

**Comment by gobengo** [comment]
I verified the desired behaviors on the preview link (i.e. ensuring that the footer links for 'Status' and 'Open an Issue' now open in a new tab for me because now they render with attr `target=_blank`)

File: packages/website/components/link/link.js [hunk 1]
```diff
@@ -2,14 +2,61 @@ import React from 'react';
...
+ * A generic hyperlink component.
+ * * by default, links to external sites will have target=_blank if they are an external link
+ * @param {object} props
+ * @param {string} props.href - the href attribute for the link
+ * @param {number} [props.tabIndex] - the tabIndex attribute for the link
+ * @param {string} [props.target] - the target attribute for the link
+ * @param {React.ReactNode} [props.children]
+ * @param {React.MouseEventHandler<HTMLAnchorElement>} [props.onClick] - the onClick handler for the link
+ */
+const WrappedLink = ({ tabIndex = 0, href, target, ...otherProps }) => {
```
[the remark this answers is not in the document]
Comment (gobengo) [reply packages/website/components/link/link.js:48]: good catch! tysm

File: packages/website/components/link/link.js [hunk 2]
```diff
@@ -2,7 +2,59 @@ import React from 'react';
...
+      return isExternal;
+    },
+    [document]
+  );
+  return isExternalHref;
+}
+
+/**
+ * A generic hyperlink component.
+ * * by default, links to external sites will have target=_blank if they are an external link
```
Comment (yusefnapora) [inline packages/website/components/link/link.js:49]:
```suggestion
```

Comment (yusefnapora) [reply packages/website/components/link/link.js:49]: This is no longer true, right?

**Review by yusefnapora — APPROVED** [review]
Works for me 👍

**Pull request merged**